src/Sarif/Visitors/SarifVersionOneToCurrentVisitor.cs

```diff
@@ -243,21 +243,106 @@ public Rule TransformRuleVersionOne(RuleVersionOne node)
+                        {
+                            run.Files.Add(key, new FileData
+                            {
+                                Contents = new FileContent
+                                {
+                                    Text = dictionary[key]
+                                },
+                                FileLocation = fileLocation
+                            });
+                        }
+                    }
+                }
+            }
+
+            return result;
+        }
+
+        public Invocation TransformInvocationVersionOne(InvocationVersionOne node)
+        {
+            Invocation invocation = null;
+
+            if (node != null)
+            {
+                invocation = new Invocation
+                {
+                    Account = node.Account,
+                    CommandLine = node.CommandLine,
+                    EndTime = node.EndTime,
+                    EnvironmentVariables = node.EnvironmentVariables,
+                    Machine = node.Machine,
+                    ProcessId = node.ProcessId,
+                    Properties = node.Properties,
+                    ResponseFiles = CreateFileLocationsFromDictionary(node.ResponseFiles, SarifLog.Runs.Last()),
+                    StartTime = node.StartTime,
+                    WorkingDirectory = node.WorkingDirectory
+                };
+
+                if (!string.IsNullOrWhiteSpace(node.FileName))
+                {
+                    if (!string.IsNullOrWhiteSpace(node.CommandLine))
+                    {
+                        string exeName = Path.GetFileName(node.FileName);
+                        int startIndex = node.CommandLine.IndexOf(exeName) + exeName.Length;
+                        string[] args = node.CommandLine.Substring(startIndex).Split(new char[] { ' ' }, StringSplitOptions.RemoveEmptyEntries);
+                        invocation.Arguments = new List<string>(args);
+                    }
+
+                    invocation.ExecutableLocation = new FileLocation
+                    {
+                        Uri = new Uri(node.FileName, UriKind.RelativeOrAbsolute)
+                    };
+                }
+            }
+
+            return invocation;
+        }
+
         public override RunVersionOne VisitRunVersionOne(RunVersionOne node)
         {
             if (node != null)
             {
                 Run run = new Run()
                 {
                     Architecture = node.Architecture,
-                    BaselineId = node.BaselineId,
                     AutomationId = node.AutomationId,
+                    BaselineId = node.BaselineId,
                     Id = node.Id,
                     Properties = node.Properties,
                     Results = new List<Result>(),
-                    StableId = node.StableId
+                    StableId = node.StableId,
+                    Tool = TransformToolVersionOne(node.Tool)
```

Comment:
@EasyRhin0 @michaelcfanning Much better! See my mega-comment below about what this change really means.